bootstrapvz/providers/ec2/tasks/ami.py

```diff
@@ -125,3 +125,12 @@ def run(cls, info):
             registration_params['sriov_net_support'] = 'simple'
 
         info._ec2['image'] = info._ec2['connection'].register_image(**registration_params)
+
+        # Setting up tags on the AMI
+        if 'tags' in info.manifest.data and len(info.manifest.data['tags']) > 0:
```

Comment:
You can enforce the `len(info.manifest.data['tags']) > 0` by setting a min length in the manifest, this way we move as much validation as possible out of the code.